Entities/LobbyMapController.cs

```diff
@@ -0,0 +1,326 @@
+                            value.Icon = controllerInfo?.WarpIcon ?? "CollabUtils2/lobbies/warp";
+                            break;
+                        case MarkerType.RainbowBerry:
+                            value.Icon = controllerInfo?.RainbowBerryIcon ?? "CollabUtils2/lobbies/rainbowBerry";
+                            break;
+                        case MarkerType.HeartGate:
+                            value.Icon = controllerInfo?.HeartGateIcon ?? "CollabUtils2/lobbies/heartgate";
+                            break;
+                        case MarkerType.Gym:
+                            value.Icon = controllerInfo?.GymIcon ?? "CollabUtils2/lobbies/gym";
+                            break;
+                        case MarkerType.Map:
+                            value.Icon = controllerInfo?.MapIcon ?? "CollabUtils2/lobbies/map";
+                            break;
+                        case MarkerType.HeartSide:
+                            value.Icon = controllerInfo?.HeartSideIcon ?? "CollabUtils2/lobbies/heartside";
+                            break;
+                        case MarkerType.Journal:
+                            value.Icon = controllerInfo?.JournalIcon ?? "CollabUtils2/lobbies/journal";
+                            break;
+                    }
+                }
+
+                // if this marker represents an enterable map, try to read the data for that map
+                if (value.Type == MarkerType.Map && !string.IsNullOrWhiteSpace(value.Map)) {
+                    value.MapInfo = new MapInfo(value.Map);
+                }
+
+                // successfully parsed the entity
+                return true;
+            }
+        }
+
+        public struct MapInfo {
+            public readonly string SID;
+            public readonly bool Completed;
+            public readonly int Difficulty;
+            public readonly bool HeartSide;
+
+            public MapInfo(string mapName) {
+                SID = string.Empty;
+                Completed = false;
+                Difficulty = -1;
+                HeartSide = false;
+                
+                if (AreaData.Get(mapName) is AreaData areaData) {
+                    SID = areaData.SID;
+                    AreaStats areaStatsFor = SaveData.Instance.GetAreaStatsFor(areaData.ToKey());
+                    Completed = areaStatsFor != null && areaStatsFor.Modes[0].Completed;
+
+                    string mapDifficultyIconPath = areaData.Icon;
+                    if (!string.IsNullOrWhiteSpace(mapDifficultyIconPath)) {
+                        var iconFilename = mapDifficultyIconPath.Split('/').LastOrDefault() ?? string.Empty;
+                        var firstToken = iconFilename.Split('-').FirstOrDefault() ?? string.Empty;
+                        if (!string.IsNullOrWhiteSpace(firstToken)) {
+                            if (firstToken.Equals("ZZ", StringComparison.InvariantCultureIgnoreCase)) {
```

Comment:
same here for the `LobbyHelper.IsHeartSide` thing

Reply:
This can be removed since we can just check the MarkerType.